pycloudlib/lxd/instance.py

```diff
@@ -272,3 +304,33 @@ def wait_for_stop(self):
 
         Not used for LXD.
         """
+
+    def _wait_for_cloudinit(self, *, raise_on_failure: bool):
+        """Wait until cloud-init has finished.
+
+        If the instance is a virtual machine, we need to wait for the
+        lxd-agent to be ready before getting the cloud-init status.
+        To do that, we retry on vm instances if raise_on_failure
+        is specified.
+
+        :param raise_on_failure:
+            When `True`, if the process waiting for cloud-init exits non-zero
+            then this method will raise an `OSError`.
+        """
+        if self._is_vm and raise_on_failure:
```

Comment:
when lxd.get_instance is called we don't detect Type: virtual-machine on those returned LXDInstances, as a result self._is_vm is False and we fall through to the setup()._wait_for_cloud_init which fails with 
```
  File "/home/csmith/src/pycloudlib/pycloudlib/instance.py", line 392, in _wait_for_cloudinit
    raise OSError(
OSError: cloud-init failed to start: out:  error: Error: Failed to connect to lxd-agent
```